api/src/main/java/com/github/streamshub/console/api/ClientFactory.java

```diff
@@ -522,6 +562,42 @@ Map<String, Object> buildConfig(Set<String> configNames,
         return cfg;
     }
 
+    private void applyListenerConfiguration(Map<String, Object> cfg, GenericKafkaListener listener) {
+        var authType = Optional.ofNullable(listener.getAuth())
+                .map(KafkaListenerAuthentication::getType)
+                .orElse("");
+        boolean saslEnabled;
+
+        switch (authType) {
+            case "oauth":
+                saslEnabled = true;
+                cfg.putIfAbsent(SaslConfigs.SASL_MECHANISM, OAUTHBEARER);
+                cfg.putIfAbsent(SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS, STRIMZI_OAUTH_CALLBACK);
+                break;
+            case "scram-sha-512":
+                cfg.putIfAbsent(SaslConfigs.SASL_MECHANISM, SCRAM_SHA512);
+                saslEnabled = true;
+                break;
```

Comment:
Shouldn't there be a case for SCRAM SHA 256 as well?

Reply:
Strimzi doesn't support 256, so this switch will only be 512. The configuration could have specified 256 however, so I can add that below where the user/pass are taken from the request header.